src/bmpcodec.c

```diff
@@ -827,6 +839,41 @@ gdip_read_BITMAPINFOHEADER (void *pointer, ImageSource source, BITMAPV5HEADER *b
 	}
 
     return Ok;
+	
+fill_rgb_masks:
+	if (bmi->bV5BitCount != 16) {
+		bmi->bV5RedMask = 0;
+		bmi->bV5GreenMask = 0;
+		bmi->bV5BlueMask = 0;
+		return Ok;
+	}
+	if (bmi->bV5Compression != BI_BITFIELDS) {
+		// Non-compressed 16bpp bitmaps are always RGB555.
+		bmi->bV5RedMask = 0x7C00;
+		bmi->bV5GreenMask = 0x3E0;
+		bmi->bV5BlueMask = 0x1F;
+		return Ok;
+	}
```

Comment:
These two block could probably be moved up to the "goto fill_rgb_masks" point and then the mask reading code could be shared instead of being duplicated. In fact, I think the `if (bmi->bV5BitCount != 16)` block is unnecessary.